AP_Randomizer/apworld/rules_normal.py

```diff
@@ -4,8 +4,56 @@
 class PseudoregaliaNormalRules(PseudoregaliaRulesHelpers):
     def __init__(self, world) -> None:
         super().__init__(world)
+        upper_bailey = world.get_region("Upper Bailey")
 
         region_clauses = {
+            "Empty Bailey -> Upper Bailey": lambda state:
+                self.has_slide(state) and self.can_attack(state)  # going through inside building to reach the tip
+                or self.get_kicks(state, 2)
+                or self.get_kicks(state, 1) and self.knows_obscure(state),
+            # "Empty Bailey -> Castle Main": lambda state: True,
+            # "Empty Bailey -> Bailey => Theatre Pillar": lambda state: True,
+            # "Upper Bailey -> Empty Bailey": lambda state: True,
+            "Upper Bailey -> Tower Remains": lambda state:
+                self.kick_or_plunge(state, 4)
+                and (
+                    # get onto the bridge
+                    self.can_slidejump(state)
+                    or self.has_plunge(state) and self.knows_obscure(state)),
+            "Tower Remains -> Underbelly Little Guy": lambda state:
+                self.has_plunge(state),
+            "Tower Remains -> The Great Door": lambda state:
+                self.can_attack(state) and self.has_gem(state) and self.kick_or_plunge(state, 1),
+            "Theatre Main -> Keep Main": lambda state:
+                self.has_gem(state),
+            "Theatre Main -> Theatre Pillar": lambda state:
+                self.get_kicks(state, 2)
+                or self.get_kicks(state, 1) and self.has_plunge(state) and self.knows_obscure(state)
+                or self.has_gem(state),
+            "Theatre Main -> Castle => Theatre (Front)": lambda state:
+                self.has_gem(state)
+                or self.get_kicks(state, 1)
+                or self.can_slidejump(state),
+            "Bailey => Theatre Pillar -> Theatre Pillar": lambda state:
+                self.has_plunge(state)
+                or self.can_slidejump(state)
+                or self.get_kicks(state, 1) and self.can_bounce(state),
```

Comment:
route for getting to bubble with just sunsetter? should that be normal difficulty?

also I see how slidejump gets you to the other side of the room, but I don't think it helps you get up the pillar, does that matter for this region transition?

Reply:
that might have been crouch backflips before i was marking them obscore
```suggestion
            "Bailey => Theatre Pillar -> Theatre Pillar": lambda state:
                self.has_plunge(state) and self.knows_obscure(state)
                or self.can_slidejump(state)
                or self.get_kicks(state, 1) and self.can_bounce(state),
```